specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/WebApps.json

```diff
@@ -21567,6 +21567,412 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Workflow information returned.",
+            "schema": {
+              "$ref": "#/definitions/WorkflowEnvelopeCollection"
+            }
+          },
+          "404": {
+            "description": "Workflow with an ID of {workflowName} is not running.",
+            "x-ms-error-response": true
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List the workflows": {
+            "$ref": "./examples/ListWorkflows.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/workflows/{workflowName}": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Get workflow information by its ID for web site, or a deployment slot.",
+        "operationId": "WebApps_GetInstanceWorkflowSlot",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/siteNameParameter"
+          },
+          {
+            "name": "slot",
+            "in": "path",
+            "description": "Name of the deployment slot.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "workflowName",
+            "in": "path",
+            "description": "Workflow name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Workflow information returned.",
+            "schema": {
+              "$ref": "#/definitions/WorkflowEnvelope"
+            }
+          },
+          "404": {
+            "description": "Workflow with a name of {workflowName} does not exist.",
+            "x-ms-error-response": true
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GET a workflow": {
+            "$ref": "./examples/GetWorkflow.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/workflowsconfiguration/connections": {
```

Comment:
>workflowsconfiguration/connections [](http://example.com/codeflow?start=122&length=34)

This doesn't look like a valid path for getting a 'connection of a workflow configuration of a slot of a site.

It should be something like:

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/workflowConfigurations/{configuration}/connections

or

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/workflowConnections/

or


/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/workflowConfigurations/default/connections

if there's a singleton configuration

Reply:
Maybe I am just misunderstanding the intention. Do you want to say a workflowConfiguration HAS A connection? Or do you want to say there's a 'connection' that IS A workflowConfiguration?

Reply:
This is getting all of the connections in the singular configuration for all of the (multiple) workflows in the app. We originally had a few more operations planned under the /workflowsConfiguration path, but only needed this single operation.

At this point, the API path has been in production for a little over two years, so I'm not able to get it changed in the timespan of our definition review. Changing the path would be a breaking change on a GA product that would have to be carefully coordinated.

Reply:
@alexkarcher-msft So IIUC it's a 'singleton-collection of connections, associated to the (singleton-)collection of workflows'? That actually concerns me as it doesn't sound like it meets ARM requirements.

Reply:
Not quite, it's a collection of connections that are applied to every workflow.

Reply:
Since the return type of this API is also WorkflowEnvelope from ARM semantics this would be considered like a connection IS A workflowConfiguration to answer Tims question.  It seems like this is defined as a singleton GETable resource which is acceptable.

Reply:
This looks like it is implemented as a singleton instance of workflowConfigurations which is fine I think. I have a few other comments about this resource though below